shared/kubernetes/provider.go

```diff
@@ -607,6 +610,10 @@ func (provider *Provider) ApplyMizuTapperDaemonSet(ctx context.Context, namespac
 	noScheduleToleration.WithOperator(core.TolerationOpExists)
 	noScheduleToleration.WithEffect(core.TaintEffectNoSchedule)
 
+	procfsVolume := applyconfcore.Volume()
```

Comment:
```suggestion
      // Host machine's `/proc` directory is needed inside the container because ...
	procfsVolume := applyconfcore.Volume()
```

Reply:
Also I guess we need to add [`readOnlyRootFilesystem`](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems). Because people might interpret mounting host's `/proc` as a security violation. Even reads might be considered as a security violation like accessing something other than the network traffic. But I'm not sure...

Reply:
Maybe we can mount only the paths directly required by the tapper and not the whole `/proc` directory.

Reply:
Done

Reply:
@dvdlevanon OK comment part is done.

But my doubt about the need of exposing whole `/proc` directory of host machine remains.

Reply:
Its very common to map procfs for privileged containers that monitor the system - such as the tapper. In the future we might need to map sysfs, devfs and such.

Regarding `readOnlyRootFilesystem`, good idea, i'll do it

Reply:
Read only done @mertyildiran

Reply:
For the record, I'm pasting what I said previously and resolving the thread;
> For the procfs, let me give you an example from the user's point of view;
_I've some process that writes sensitive information to `stdout`. I see Mizu has access to the file descriptors of all processes. I cannot be sure whether I'm leaking some sensitive information to Mizu without reviewing its source code. It's better to have access only the necessary information._